nf_core/lint.py

```diff
@@ -543,6 +570,39 @@ def check_conda_dockerfile(self):
                 if not found_strings[idx]:
                     self.failed.append((9, "Could not find Dockerfile string: {}".format(s)))
 
+
+    def check_conda_singularityfile(self):
+        """ Check that the Singularity build file looks right, if working with conda
+
+        Make sure that a name is given and is consistent with the pipeline name
+        Check that depedency versions are pinned
+        Warn if dependency versions are not the latest available """
+
+        if 'environment.yml' not in self.files or len(self.singularityfile) == 0:
+            return
+
+        expected_strings = [
+            'From:nfcore/base',
+            'Bootstrap:docker',
+            'VERSION {}'.format(self.config['params.version'].strip(' \'"')),
+            'PATH=/opt/conda/envs/{}/bin:$PATH'.format(self.conda_config['name']),
+            'export PATH',
+            'environment.yml /',
+            '/opt/conda/bin/conda env create -f /environment.yml',
+            '/opt/conda/bin/conda clean -a',
+        ]
+        found_strings = [False for x in expected_strings]
+        for l in self.singularityfile:
+            for idx, s in enumerate(expected_strings):
+                if l.strip() == s.strip():
+                    found_strings[idx] = True
+        if all(found_strings):
+            self.passed.append((10, "Found all expected strings in Singularity file"))
+        else:
+            for idx, s in enumerate(expected_strings):
+                if not found_strings[idx]:
+                    self.failed.append((10, "Could not find Singularity file string: {}".format(s)))
+
```

Comment:
All correct here. Just an alternative avoiding nested loops, but really, just an idea, not required:

```
difference = set(expected_strings) - set(self.singularity_file)
if not difference:
    self.passed.append((10, "Found all expected strings in Singularity file"))
else:
   for missing in difference:
        self.failed.append((10, "Could not find Singularity file string: {}".format(missing)))
``